File: ui/etcd-ui/src/api/connection.js
import request from "@/utils/request"


export function createConnection(data) {
    return request({
        url: "/v1/connections",
        method: "post",
        data: data
    })
}

export function listConnection(param) {
    return request({
        url: `/v1/connections?page=${param.page}&page_size=${param.pageSize}&name=${param.name}`,
        method: "get"
    })

}

export function getConnection(id) {
    return request({
        url: `/v1/connections/${id}`,
        method: "get"
    })
}

export function delConnection(id) {
    return request({
        url: `/v1/connections/${id}`,
        method: "delete"
    })
}

export function updateConnection(data) {
    return request({
        url: `/v1/connections/${data.id}`,
        method: "put",
        data: data
    })
}

File: backend/route/connection.go
package Route

import (
	"github.com/gin-gonic/gin"

	"github.com/wanyuqin/etcd-ui/backend/internal/handler"
)

func ConnectionRoute(r *gin.RouterGroup) {
	v1 := r.Group("/connections")
	{
		v1.POST("", handler.CreateConnection)
		v1.GET("", handler.ListConnection)
		v1.PUT("/:id", handler.UpdateConnection)
		v1.GET("/:id", handler.GetConnection)
		v1.DELETE("/:id", handler.DeleteConnection)
		v1.POST("/init", handler.InitConnection)
	}
}
